integrations/repo_migrate_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+//+build extra
+
+package integrations
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func testRepoMigrate(t *testing.T, session *TestSession, cloneAddr string) *TestResponse {
+	req := NewRequest(t, "GET", "/repo/migrate")
+	resp := session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusOK, resp.HeaderCode)
+	htmlDoc, err := NewHtmlParser(resp.Body)
+	assert.NoError(t, err)
+
+	link, exists := htmlDoc.doc.Find("form.ui.form").Attr("action")
+	assert.True(t, exists, "The template has changed")
+
+	uid, exists := htmlDoc.doc.Find("#uid").Attr("value")
+	assert.True(t, exists, "The template has changed")
+
+	repoDirname := path.Base(cloneAddr)
+	repoName := repoDirname[0 : len(repoDirname)-len(path.Ext(repoDirname))]
+
+	fmt.Fprintf(os.Stderr, "cloneAddr:%s uid:%s repoName:%s\n", cloneAddr, uid, repoName)
+	req = NewRequestBody(t, "POST", link,
+		bytes.NewBufferString(url.Values{
+			"_csrf":      []string{htmlDoc.GetInputValueByName("_csrf")},
+			"clone_addr": []string{cloneAddr},
+			"uid":        []string{uid},
+			"repo_name":  []string{repoName},
+		}.Encode()),
+	)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	resp = session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusFound, resp.HeaderCode)
+
+	return resp
+}
+
+func TestRepoMigrate(t *testing.B) {
+	prepareTestEnv(t)
+	session := loginUser(t, "user2", "password")
+	testRepoMigrate(t, session, "git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git")
```

Comment:
We could maybe test with a smaller repo (like gitea ^^) and benchmark with biggest one  ?